src/main/java/de/tum/in/www1/artemis/service/connectors/hades/HadesCIService.java

```diff
@@ -0,0 +1,165 @@
+package de.tum.in.www1.artemis.service.connectors.hades;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_HADES;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.domain.VcsRepositoryUri;
+import de.tum.in.www1.artemis.domain.enumeration.ProgrammingLanguage;
+import de.tum.in.www1.artemis.domain.participation.ProgrammingExerciseParticipation;
+import de.tum.in.www1.artemis.exception.ContinuousIntegrationException;
+import de.tum.in.www1.artemis.service.connectors.ConnectorHealth;
+import de.tum.in.www1.artemis.service.connectors.ci.AbstractContinuousIntegrationService;
+import de.tum.in.www1.artemis.service.connectors.ci.CIPermission;
+import de.tum.in.www1.artemis.service.connectors.hades.dto.HadesBuildResultNotificationDTO;
+import de.tum.in.www1.artemis.web.rest.dto.CheckoutDirectoriesDTO;
+
+@Service
+@Profile(PROFILE_HADES)
+public class HadesCIService extends AbstractContinuousIntegrationService {
+
+    private static final Logger log = LoggerFactory.getLogger(HadesCIService.class);
+
+    private final ObjectMapper objectMapper;
+
+    private final RestTemplate shortTimeoutRestTemplate;
+
+    @Value("${artemis.continuous-integration.url}")
+    private String hadesServerUrl;
+
+    public HadesCIService(ObjectMapper objectMapper, @Qualifier("shortTimeoutHadesRestTemplate") RestTemplate shortTimeoutRestTemplate) {
+        this.objectMapper = objectMapper;
+        this.shortTimeoutRestTemplate = shortTimeoutRestTemplate;
+    }
+
+    @Override
+    public void createBuildPlanForExercise(ProgrammingExercise exercise, String planKey, VcsRepositoryUri repositoryUri, VcsRepositoryUri testRepositoryUri,
+            VcsRepositoryUri solutionRepositoryUri) {
+
+    }
+
+    @Override
+    public void recreateBuildPlansForExercise(ProgrammingExercise exercise) {
+    }
+
+    @Override
+    public String copyBuildPlan(ProgrammingExercise sourceExercise, String sourcePlanName, ProgrammingExercise targetExercise, String targetProjectName, String targetPlanName,
+            boolean targetProjectExists) {
+        return targetExercise.getProjectKey() + "-" + targetPlanName.toUpperCase().replaceAll("[^A-Z0-9]", "");
```

Comment:
Maybe I do not fully understand this line 😅 When removing all letters and numbers from the `targetPlanName`, why do you even add it at all? What's the reason for this naming convention?

Reply:
This removes everything except letters and numbers. `[^X]` means that you target everything except X